### DISCUSSION TITLE
How to load TImage32 from a TMemoryStream

### DISCUSSION THREAD

**Discussion by chicknsoup** [AngusJohnson/Image32#42] [General]
Hi,
I can load the image via TFileStream but not TMemoryStream. 
I don't want to save the image to disk and use LoadFromFile procedure.
Is there a way to directly load the image from memory? 

Thank you.

**Comment by AngusJohnson** [comment]
```
uses
  Img32, Img32.Fmt.BMP;

procedure TForm1.PaintBox1Paint(Sender: TObject);
var
  ms: TMemoryStream;
  img: TImage32;
const
  margin = 20;
begin
  ms := TMemoryStream.Create;
  img := TImage32.create;
  try
    ms.LoadFromFile('../../beetle.bmp');
    img.LoadFromStream(ms);
    PaintBox1.SetBounds(margin, margin, img.Width + margin, img.Height + margin);
    img.CopyToDc(PaintBox1.Canvas.Handle);
  finally
    img.Free;
    ms.Free;
  end;
end;
```

[Image32_sample.zip](https://github.com/AngusJohnson/Image32/files/13799889/Image32_sample.zip)

**Reply by chicknsoup** [reply]
Thank you, but let me clarify.
I have a resource file (.res) which contains multiple images. That file will be loaded and parsed in runtime using TResModule (https://github.com/zuloloxi/XN-Resource-Editor/blob/master/source/runtime/unitResFile.pas). I tried img.LoadFromStream with ResModule.ResourceDetails[0].Data (which is a TMemoryStream) and it didn't work. (Saving ResModule.ResourceDetails[0].Data to a file and then loading it with img.LoadFromFile still works).

**Reply by chicknsoup** [reply]
I tried the following:
1. Create and img: TImage32 and LoadFromFile('file.png') --> OK
2. Create a ms: TMemoryStream
3. Call img.SaveToStream(ms) --> OK, the stream contains all required data
4. Reload img with img.LoadFromStream(ms) --> NOK

So there could be something wrong with the LoadFromStream procedure (or it should not work with a TMemoryStream by default?)

**Comment by AngusJohnson** [comment]
https://github.com/AngusJohnson/Image32/blob/b431b07407f326de4cc73ff26af0878786671cf4/source/Img32.pas#L2507

**Reply by chicknsoup** [reply]
But the .res file is loaded during runtime (not compile time), so LoadFromResource function will not work and raise an error: Resource not found.

**Comment by chicknsoup** [comment]
Okay, I found the culprit. We have to reset the stream position to 0 before loading it using img.LoadFromStream. My bad :(
